### PR TITLE
User service extended for register

### PR DESCRIPTION
**Pull request by angeles-tellaarena**
Esta la parte de register, falta la parte de login.
un par de cosas que hay que arreglar:
user se persiste bien pero las cosas que crea (vehiculo, pasajero, etc) tienen id null.
no se muestra el mail del usuario en la barra lateral.

### CODE REVIEW

**Comment by angeles-tellaarena** [comment]
@camilagarcia113 @javierperini

File: desapp-groupb-frontend/scripts/app.js [hunk 1]
```diff
@@ -49,11 +51,14 @@ app.controller('HomeController', function ($scope, $http, $translate) {
       $scope.register = true;
    }
 
+   $scope.loginUser=function(user){
+      $http.get($scope.baseUrl + 'users/' + user.email + '/' + user.password)
```
Comment (camilagarcia113) [inline desapp-groupb-frontend/scripts/app.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2587695756/issues/source?bid=3284218&fileBranchId=3305265#l55)

Comment (camilagarcia113) [inline desapp-groupb-frontend/scripts/app.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2587695756/issues/source?bid=3284218&fileBranchId=3305265#l55)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/services/UserService.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import javax.ws.rs.*;
+
+@Path("/users")
+public class UserService extends GenericService<User> {
+
+    @GET
+    @Path("/register/{email}/{password}")
+    @Produces("application/json")
+    @Transactional
+    public User forRegistration(@PathParam("email") String email, @PathParam("password") String password){
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/services/UserService.java]: Codacy Issue found: [Line is longer than 80 characters (found 106).](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2587695800/issues/source?bid=3284218&fileBranchId=3305265#l15)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 3]
```diff
@@ -9,192 +9,211 @@
...
 
     public void sendPrivateMessageTo(User receiver, String message) {
         PrivateMessage privateMessage = new PrivateMessage(this, receiver, message);
         receiver.receiveMessage(privateMessage);
     }
 
     public void receiveMessage(Message message){
-        messages.add(message);
-    }
+          messages.add(message);
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 10, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2587695779/issues/source?bid=3284218&fileBranchId=3305265#l54)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 4]
```diff
@@ -9,192 +9,211 @@
...
+    public String getEmail() {
+      return email;
+    }
+
+    public void setEmail(String email) {
+      this.email = email;
+    }
+
+    public String getPassword() {
+      return password;
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2587695779/issues/source?bid=3284218&fileBranchId=3305265#l213)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 5]
```diff
@@ -9,192 +9,211 @@
...
-	}
+    public Role getCurrentRole() {
+      return currentRole;
+    }
 
-	public void setPassengerRole(Passenger passengerRole) {
-		this.passengerRole = passengerRole;
-	}
+    public void setCurrentRole(Role currentRole) {
+      this.currentRole = currentRole;
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2587695779/issues/source?bid=3284218&fileBranchId=3305265#l144)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 6]
```diff
@@ -9,192 +9,211 @@
...
-	public void setPoints(int points) {
-		this.points = points;
-	}
+    @Transient
+    public Passenger getPassengerRole() {
+      return passengerRole;
+    }
+
+    public void setPassengerRole(Passenger passengerRole) {
+      this.passengerRole = passengerRole;
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2587695779/issues/source?bid=3284218&fileBranchId=3305265#l135)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 7]
```diff
@@ -9,192 +9,211 @@
...
-	public void setDriverRole(Driver driverRole) {
-		this.driverRole = driverRole;
-	}
+    public void setDriverRole(Driver driverRole) {
+      this.driverRole = driverRole;
+    }
 
     public int getPoints(){
-        return points;
+          return points;
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 10, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2587695779/issues/source?bid=3284218&fileBranchId=3305265#l122)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 8]
```diff
@@ -9,192 +9,211 @@
...
+    public void setEmail(String email) {
+      this.email = email;
+    }
+
+    public String getPassword() {
+      return password;
+    }
+
+    public void setPassword(String password) {
+      this.password = password;
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2587695779/issues/source?bid=3284218&fileBranchId=3305265#l217)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 9]
```diff
@@ -9,192 +9,211 @@
...
+      return currentRole.driver();
+    }
 
-	@Transient
-	public Driver getDriverRole() {
-		return driverRole;
-	}
+    @Transient
+    public Driver getDriverRole() {
+      return driverRole;
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2587695779/issues/source?bid=3284218&fileBranchId=3305265#l114)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java (hunk 3 above)
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 10, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2594143021/issues/source?bid=3284218&fileBranchId=3305265#l54)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java (hunk 4 above)
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2594143021/issues/source?bid=3284218&fileBranchId=3305265#l213)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java (hunk 5 above)
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2594143021/issues/source?bid=3284218&fileBranchId=3305265#l144)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java (hunk 6 above)
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2594143021/issues/source?bid=3284218&fileBranchId=3305265#l135)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java (hunk 7 above)
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 10, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2594143021/issues/source?bid=3284218&fileBranchId=3305265#l122)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java (hunk 8 above)
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2594143021/issues/source?bid=3284218&fileBranchId=3305265#l217)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java (hunk 9 above)
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2594143021/issues/source?bid=3284218&fileBranchId=3305265#l114)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 10]
```diff
@@ -9,192 +9,211 @@
...
+        this.messages = messages;
+    }
+
+    @Transient
+    public int getCapacityVehicle() {
       return vehicle.getCapacity();
   }
+
+    public String getEmail() {
+      return email;
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2594143021/issues/source?bid=3284218&fileBranchId=3305265#l205)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 11]
```diff
@@ -9,192 +9,211 @@
...
+      this.currentRole = currentRole;
+    }
+
+    @Id @GeneratedValue
+    public Integer getId() {
+      return id;
+    }
+
+    public void setId(Integer id) {
+      this.id = id;
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2594143021/issues/source?bid=3284218&fileBranchId=3305265#l153)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 12]
```diff
@@ -9,192 +9,211 @@
...
+    public int getCapacityVehicle() {
       return vehicle.getCapacity();
   }
+
+    public String getEmail() {
+      return email;
+    }
+
+    public void setEmail(String email) {
+      this.email = email;
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2594143021/issues/source?bid=3284218&fileBranchId=3305265#l209)

File: desapp-groupb-frontend/scripts/controllers/maps.js [hunk 13]
```diff
@@ -1,28 +1,30 @@
...
-    var buenos_aires = {
+    var originPlaceId = null;
+    var destinationPlaceId = null;
+    var travelMode = google.maps.TravelMode.DRIVING;
+    var buenosAires = {
         lat : -34.6261939,
         lng : -58.410998
     };
-    var map = new google.maps.Map(document.getElementById('map'), {
+    var map = new google.maps.Map(document.getElementById("map"), {
```
Comment (camilagarcia113) [inline desapp-groupb-frontend/scripts/controllers/maps.js]: Codacy Issue found: ["google" is not defined.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2595634836/issues/source?bid=3284218&fileBranchId=3305265#l11)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java (hunk 3 above)
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 10, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2595634856/issues/source?bid=3284218&fileBranchId=3305265#l54)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java (hunk 4 above)
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2595634856/issues/source?bid=3284218&fileBranchId=3305265#l213)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java (hunk 5 above)
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2595634856/issues/source?bid=3284218&fileBranchId=3305265#l144)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java (hunk 6 above)
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2595634856/issues/source?bid=3284218&fileBranchId=3305265#l135)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java (hunk 7 above)
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 10, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2595634856/issues/source?bid=3284218&fileBranchId=3305265#l122)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java (hunk 8 above)
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2595634856/issues/source?bid=3284218&fileBranchId=3305265#l217)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java (hunk 9 above)
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2595634856/issues/source?bid=3284218&fileBranchId=3305265#l114)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java (hunk 11 above)
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2595634856/issues/source?bid=3284218&fileBranchId=3305265#l153)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java (hunk 10 above)
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2595634856/issues/source?bid=3284218&fileBranchId=3305265#l205)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 14]
```diff
@@ -3,198 +3,235 @@
...
-  public List<Message> getMessages(){
-      return messages;
-  }
+    public List<Route> getRoutes(){
+      return driverRole.getRoutes();
+    }
 
-  public void setMessages(List<Message> messages) {
-      this.messages = messages;
-  }
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def modifier' have incorrect indentation level 6, expected level should be 4.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2612346144/issues/source?bid=3284218&fileBranchId=3305265#l179)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 15]
```diff
@@ -3,198 +3,235 @@
...
+        points -= product.getCost() * quantity;
+        product.subtractStock(quantity);
     }
 
-	@Transient
-	public boolean isPassengerRoleActivated() {
-		return currentRole.passenger();
-	}
+    @Transient
+    public boolean isPassengerRoleActivated() {
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2612346144/issues/source?bid=3284218&fileBranchId=3305265#l110)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 16]
```diff
@@ -3,198 +3,235 @@
...
+      return currentRole.passenger();
+    }
 
     @Transient
-	public boolean isDriverRoleActivated() {
-		return currentRole.driver();
-	}
+    public boolean isDriverRoleActivated() {
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2612346144/issues/source?bid=3284218&fileBranchId=3305265#l115)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 17]
```diff
@@ -3,198 +3,235 @@
...
 
-	public void setDriverRole(Driver driverRole) {
-		this.driverRole = driverRole;
-	}
+    public void setDriverRole(Driver driverRole) {
+      this.driverRole = driverRole;
+    }
 
     public int getPoints(){
-        return points;
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 10, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2612346144/issues/source?bid=3284218&fileBranchId=3305265#l129)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 18]
```diff
@@ -3,198 +3,235 @@
...
+    public Role getCurrentRole() {
+      return currentRole;
+    }
+
+    public void setCurrentRole(Role currentRole) {
+      this.currentRole = currentRole;
+    }
+
+    @Id @GeneratedValue
+    public Integer getId() {
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2612346144/issues/source?bid=3284218&fileBranchId=3305265#l157)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 19]
```diff
@@ -3,198 +3,235 @@
...
 
     @Transient
-	public Passenger getPassengerRole() {
-		return passengerRole;
-	}
+    public Role getCurrentRole() {
+      return currentRole;
+    }
+
+    public void setCurrentRole(Role currentRole) {
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2612346144/issues/source?bid=3284218&fileBranchId=3305265#l152)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 20]
```diff
@@ -3,198 +3,235 @@
...
+    public void setDriverRole(Driver driverRole) {
+      this.driverRole = driverRole;
+    }
 
     public int getPoints(){
-        return points;
+          return points;
+      }
+
+    public void setPoints(int points) {
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2612346144/issues/source?bid=3284218&fileBranchId=3305265#l133)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 21]
```diff
@@ -3,198 +3,235 @@
...
-	}
+    public void setPassengerRole(Passenger passengerRole) {
+      this.passengerRole = passengerRole;
+    }
 
     @Transient
-	public Passenger getPassengerRole() {
-		return passengerRole;
-	}
+    public Role getCurrentRole() {
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2612346144/issues/source?bid=3284218&fileBranchId=3305265#l148)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 22]
```diff
@@ -3,198 +3,235 @@
...
+          return points;
+      }
+
+    public void setPoints(int points) {
+      this.points = points;
+    }
+
+    @OneToOne
+    @Cascade(org.hibernate.annotations.CascadeType.SAVE_UPDATE)
+    public Passenger getPassengerRole() {
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2612346144/issues/source?bid=3284218&fileBranchId=3305265#l139)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 23]
```diff
@@ -3,198 +3,235 @@
...
-      return messages;
-  }
+    public List<Route> getRoutes(){
+      return driverRole.getRoutes();
+    }
 
-  public void setMessages(List<Message> messages) {
-      this.messages = messages;
-  }
+      public void setRoutes(List<Route> routes){
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 10, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2612346144/issues/source?bid=3284218&fileBranchId=3305265#l180)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/Passenger.java [hunk 24]
```diff
@@ -8,6 +8,8 @@
 @Entity
 public class Passenger extends Role {
 
+    public Passenger(){}
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/Passenger.java]: Codacy Issue found: [Avoid unnecessary constructors - the compiler will generate these for you](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2612422798/issues/source?bid=3284218&fileBranchId=3305265#l11)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/services/RideRequestService.java [hunk 25]
```diff
@@ -0,0 +1,38 @@
...
+    private RideRepository rideRepository;
+    @Autowired
+    private UserRepository userRepository;
+
+    @POST
+    @Path("/joinRide")
+    @Consumes("application/json")
+    @Transactional
+    public RideRequest joinRide(Ride ride, User user){
+        RideRequest rideRequest = new RideRequest(user, rideRepository.find(ride.getId()), new DayOfWeekRideDate(DateTimeConstants.TUESDAY));
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/services/RideRequestService.java]: Codacy Issue found: [Line is longer than 80 characters (found 141).](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2612422831/issues/source?bid=3284218&fileBranchId=3305265#l30)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 26]
```diff
@@ -3,198 +3,236 @@
 import java.util.ArrayList;
 import java.util.List;
 import ar.edu.unq.desapp.grupob.model.exceptions.RideRequestException;
+import org.hibernate.annotations.*;
+import org.springframework.transaction.annotation.Transactional;
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: [Avoid unused imports such as 'org.springframework.transaction.annotation.Transactional'](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2612422812/issues/source?bid=3284218&fileBranchId=3305265#l7)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 27]
```diff
@@ -3,198 +3,236 @@
...
-	}
+    public void setPassengerRole(Passenger passengerRole) {
+      this.passengerRole = passengerRole;
+    }
 
     @Transient
-	public Passenger getPassengerRole() {
-		return passengerRole;
-	}
+    public Role getCurrentRole() {
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2612422812/issues/source?bid=3284218&fileBranchId=3305265#l149)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 28]
```diff
@@ -3,198 +3,236 @@
...
+        points -= product.getCost() * quantity;
+        product.subtractStock(quantity);
     }
 
-	@Transient
-	public boolean isPassengerRoleActivated() {
-		return currentRole.passenger();
-	}
+    @Transient
+    public boolean isPassengerRoleActivated() {
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2612422812/issues/source?bid=3284218&fileBranchId=3305265#l110)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 29]
```diff
@@ -3,198 +3,236 @@
...
+      return currentRole.passenger();
+    }
 
     @Transient
-	public boolean isDriverRoleActivated() {
-		return currentRole.driver();
-	}
+    public boolean isDriverRoleActivated() {
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2612422812/issues/source?bid=3284218&fileBranchId=3305265#l115)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 30]
```diff
@@ -3,198 +3,236 @@
...
-  public List<Message> getMessages(){
-      return messages;
-  }
+    public List<Route> getRoutes(){
+      return driverRole.getRoutes();
+    }
 
-  public void setMessages(List<Message> messages) {
-      this.messages = messages;
-  }
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def modifier' have incorrect indentation level 6, expected level should be 4.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2612422812/issues/source?bid=3284218&fileBranchId=3305265#l180)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 31]
```diff
@@ -3,198 +3,236 @@
...
+
+    public void setEmail(String email) {
+      this.email = email;
+    }
+
+    public String getPassword() {
+      return password;
+    }
+
+    public void setPassword(String password) {
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2612422812/issues/source?bid=3284218&fileBranchId=3305265#l236)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 32]
```diff
@@ -3,198 +3,236 @@
...
-      return messages;
-  }
+    public List<Route> getRoutes(){
+      return driverRole.getRoutes();
+    }
 
-  public void setMessages(List<Message> messages) {
-      this.messages = messages;
-  }
+      public void setRoutes(List<Route> routes){
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 10, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2612422812/issues/source?bid=3284218&fileBranchId=3305265#l181)

File: desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java [hunk 33]
```diff
@@ -3,198 +3,236 @@
...
 
     @Transient
-	public Passenger getPassengerRole() {
-		return passengerRole;
-	}
+    public Role getCurrentRole() {
+      return currentRole;
+    }
+
+    public void setCurrentRole(Role currentRole) {
```
Comment (camilagarcia113) [inline desapp-groupb-backend/src/main/java/ar/edu/unq/desapp/grupob/model/User.java]: Codacy Issue found: ['method def' child have incorrect indentation level 6, expected level should be 8.](https://www.codacy.com/app/camilagarcia-113/DesarrolloDeAplicaciones-GrupoB/file/2612422812/issues/source?bid=3284218&fileBranchId=3305265#l153)

**Pull request merged**